src/main/java/strategy/OperationStrategyFactory.java

```diff
@@ -0,0 +1,21 @@
+package strategy;
+
+import java.util.HashMap;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class OperationStrategyFactory {
+    private final Map<FruitTransaction.Operation, OperationStrategy> strategies;
+
+    public OperationStrategyFactory() {
+        strategies = new HashMap<>();
+        strategies.put(FruitTransaction.Operation.BALANCE, new BalanceOperationStrategy());
+        strategies.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationStrategy());
+        strategies.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperationStrategy());
+        strategies.put(FruitTransaction.Operation.RETURN, new ReturnOperationStrategy());
```

Comment:
And implementation provided is already not strategy. So better to change naming to represent what it's doing
some possible examples (but choose same naming pattern for all of them)
```suggestion
        strategies.put(FruitTransaction.Operation.BALANCE, new BalanceOperationExecutor());
        strategies.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler());
        strategies.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperationFacilitator());
        strategies.put(FruitTransaction.Operation.RETURN, new ReturnOperationPerformer());
```